patches/nginx-1.11.2-delayed_posted_events.patch

```diff
@@ -0,0 +1,98 @@
+diff --git a/src/event/ngx_event.c b/src/event/ngx_event.c
+index c8ae5b2..948760e 100644
+--- a/src/event/ngx_event.c
++++ b/src/event/ngx_event.c
+@@ -193,8 +193,10 @@ ngx_module_t  ngx_event_core_module = {
+ void
+ ngx_process_events_and_timers(ngx_cycle_t *cycle)
+ {
+-    ngx_uint_t  flags;
+-    ngx_msec_t  timer, delta;
++    ngx_uint_t   flags;
++    ngx_msec_t   timer, delta;
++    ngx_queue_t *q;
++    ngx_event_t *ev;
+ 
+     if (ngx_timer_resolution) {
+         timer = NGX_TIMER_INFINITE;
+@@ -215,6 +217,13 @@ ngx_process_events_and_timers(ngx_cycle_t *cycle)
+ #endif
+     }
+ 
++    if (!ngx_queue_empty(&ngx_posted_delayed_events)) {
++        ngx_log_debug0(NGX_LOG_DEBUG_EVENT, cycle->log, 0,
++                       "posted delayed event queue not empty"
++                       " making poll timeout 0");
++        timer = 0;
++    }
++
+     if (ngx_use_accept_mutex) {
+         if (ngx_accept_disabled > 0) {
+             ngx_accept_disabled--;
+@@ -257,6 +266,32 @@ ngx_process_events_and_timers(ngx_cycle_t *cycle)
+     }
+ 
+     ngx_event_process_posted(cycle, &ngx_posted_events);
++
++    while (!ngx_queue_empty(&ngx_posted_delayed_events)) {
++        q = ngx_queue_head(&ngx_posted_delayed_events);
++
++        ev = ngx_queue_data(q, ngx_event_t, queue);
++        if (ev->delayed) {
++            /* start of newly inserted nodes */
++            for (; q != ngx_queue_sentinel(&ngx_posted_delayed_events);
```

Comment:
Better put the stuff after `;` in a separate line to make the code align up more nicely?